### ISSUE TITLE
Copyright Year out of date

### ISSUE THREAD

**Issue by kodebach** [ElektraInitiative/libelektra#3681]
The year in LICENSE.md still says 2018

https://github.com/ElektraInitiative/libelektra/blob/758d002d44764dabc8cdce4635de8285a5f99e00/LICENSE.md#L1-L3

There may also be other locations that are outdated.

**Comment by markus2330** [comment]
Thank you for pointing out! It should read `2004-2021`

**Comment by a-kraschitzer** [comment]
I would like to take this over

**Comment by a-kraschitzer** [comment]
doing a quick git grep showed me these further occurrences:
`git grep "Copyright (.) .* Elektra"`
```
src/libs/version_gen.awk:# Copyright (C) 2019 Elektra Initiative.
src/tools/kdb/gen/templates/collect.sh:# Copyright (C) 2019 Elektra Initiative.
src/tools/kdb/gen/templates/highlevel.c.mustache: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
src/tools/kdb/gen/templates/highlevel.commands.h.mustache: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
src/tools/kdb/gen/templates/highlevel.h.mustache: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
src/tools/kdb/gen/templates/highlevel/LICENSE.txt:Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/commands.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/commands.expected.commands.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/commands.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/empty.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/empty.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/enum.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/enum.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/externalspec.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/externalspec.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/externalwithdefaults.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/externalwithdefaults.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/nosetter.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/nosetter.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/notype.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/notype.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/simple.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/simple.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/struct.expected.c: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
tests/shell/gen/highlevel/struct.expected.h: *     Copyright (C) 2019 Elektra Initiative (https://libelektra.org)
```

should they all be updated?

**Comment by a-kraschitzer** [comment]
I also found these occurrences which don't say `Elektra Initiative`
`git grep "Copyright (.) [0-9-]* [^E]"`
```
scripts/cmake/Modules/DownloadProjectLicense.txt:Copyright (c) 2015 Crascit
scripts/cmake/Modules/FindLibGpgme.cmake:# Copyright (c) 2018 Peter Nirschl.
scripts/cmake/Modules/FindLibOpenSSL.cmake:# Copyright (c) 2017 Peter Nirschl <peter.nirschl@gmail.com>
scripts/dev/realpath:# Copyright (c) 2014 Michael Kropat
src/libs/version_gen.awk:# Copyright (C) 2006 Daniel M. Eischen.  All rights reserved.
src/plugins/ni/nickel-1.1.0/COPYING:Copyright (C) 2008 Charles Lindsay
src/plugins/ni/nickel-1.1.0/README:Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/include/bohr/ni.h: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/Makefile:# Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/buf.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/hash.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/include/bohr/ds_hash.h: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/include/bohr/ds_str.h: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/include/bohr/ds_vector.h: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/internal.h: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/io.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/nickel.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/nickel-1.1.0/src/tests/ni.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/plugins/ni/testmod_ni.c: * Copyright (C) 2008 Charles Lindsay.  Some rights reserved; see COPYING.
src/tools/website/public/assets/fonts/lato/OFL.txt:Copyright (c) 2010-2014 by tyPoland Lukasz Dziedzic (team@latofonts.com) with Reserved Font Name "Lato"
src/tools/website/public/vendor/asciinema/asciinema-player.js: * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```

**Comment by kodebach** [comment]
> should they all be updated?

- `src/libs/version_gen.awk`: The copyright should probably be changed into something similar to what we use in most C files:
https://github.com/ElektraInitiative/libelektra/blob/d4ee256e66a8d2c3d630ad64a838a9c69651d3de/src/libs/elektra/backend.c#L6
Then we don't need to update it all the time.
- `src/tools/kdb/gen/templates/*`: All these files should be updated yes. They explicitly mention the copyright details, because these are templates and will be copied into application code, if used.
- `tests/shell/gen/highlevel/*`: All these files are test-data and generated from the templates above. Therefore, for the tests to succeed their copyright must match that of the templates.

We should probably also document somewhere, which files contain explicit copyright details. @markus2330 @mpranj Any suggestions for where to put this info?

> I also found these occurrences which don't say `Elektra Initiative`

These copyrights all refer to 3rd party code (most of the time the whole file is copied from another FLOSS project). The copyright should not be modified, unless we also update the other content of the file.

**Comment by markus2330** [comment]
@a-kraschitzer thank you for looking into it so carefully.

> We should probably also document somewhere, which files contain explicit copyright details. @markus2330 @mpranj Any suggestions for where to put this info?

A new section in `doc/todo/RELEASE.md`, stating what needs to be done on every first release of a new year.

**Comment by a-kraschitzer** [comment]
@markus2330 
Should I rename `doc/todo/RELEASE` to `/doc/todo/RELEASE.md` ?

**Comment by markus2330** [comment]
Yes, good idea, it is actually a Markdown file. But be beware that then the Markdown reformatter might complain.

**Comment by markus2330** [comment]
More points due to voluntary extended scope (further occurrences+renaming).

**Comment by mpranj** [comment]
Closed via #3790 (origin: #3775) thanks to @a-kraschitzer!
